### PR TITLE
*: refline logs

### PR DESCRIPTION
**Pull request by TennyZhuang**
UCP #188

### What problem does this PR solve? 

#188

### What is changed and how it works?

log behaviour

### Check List 

Tests 

 - Unit test

Code changes

### CODE REVIEW

File: cmd/cmd.go [hunk 1]
```diff
@@ -49,8 +49,8 @@ func AddFlags(cmd *cobra.Command) {
 
 	cmd.PersistentFlags().StringP(FlagLogLevel, "L", "info",
 		"Set the log level")
-	cmd.PersistentFlags().String(FlagLogFile, "",
-		"Set the log file path. If not set, logs will output to stdout")
+	cmd.PersistentFlags().String(FlagLogFile, "/tmp/br.log",
```
Comment (GregoryIan) [inline cmd/cmd.go]: It is best to output each backup to a separate log file, and please use `os.TempDir()`

The following method may be useful to you.
```
func TimestampLogFileName() string {
	return fmt.Sprintf("%s/br-%s", os.TempDir(), time.Now().Format(time.RFC3339))
}
```

File: pkg/task/backup.go [hunk 2]
```diff
@@ -213,6 +213,7 @@ func RunBackup(c context.Context, g glue.Glue, cmdName string, cfg *BackupConfig
 
 	err = client.SaveBackupMeta(ctx, ddlJobs)
 	if err != nil {
+		log.Error("save backup meta failed", zap.Reflect("meta", ddlJobs))
```
Comment (GregoryIan) [inline pkg/task/backup.go]: The number of ddl jobs may be particularly large, especially as the system runs longer and longer, it is recommended to print only version information

**Comment by GregoryIan** [comment]
@TennyZhuang Great Job.

File: pkg/summary/collector.go [hunk 3]
```diff
@@ -40,7 +40,15 @@ type LogCollector interface {
...
 
-var collector = newLogCollector(log.Info)
+var collector = func() LogCollector {
+	conf := new(log.Config)
+	// Always output summary to stdout.
+	logger, _, err := log.InitLogger(conf)
+	if err != nil {
+		logger = log.L()
+	}
+	return newLogCollector(logger.Info)
```
Comment (GregoryIan) [inline pkg/summary/collector.go]: I think we also need to output it into log file if user specify log file

Comment (TennyZhuang) [reply pkg/summary/collector.go]: It's difficult to know whether the global logger will output to stdout or output to a file here.

File: cmd/cmd.go [hunk 4]
```diff
@@ -41,6 +44,10 @@ const (
 	flagVersionShort = "V"
 )
 
+func timestampLogFileName() string {
+	return fmt.Sprintf("%s/br-%s", os.TempDir(), time.Now().Format(time.RFC3339))
```
Comment (kennytm) [inline cmd/cmd.go]:
```suggestion
	return filepath.Join(os.TempDir(), "br-" + time.Now().Format(time.RFC3339))
```

File: pkg/task/common.go [hunk 5]
```diff
@@ -268,6 +270,7 @@ func ReadBackupMeta(
 	}
 	backupMeta := &backup.BackupMeta{}
 	if err = proto.Unmarshal(metaData, backupMeta); err != nil {
+		log.Error("parse backupmeta failed", zap.Error(err), zap.Binary("meta", metaData))
```
Comment (kennytm) [inline pkg/task/common.go]: The metadata file can be huge. I don't suggest including this into the log.

File: pkg/task/restore.go [hunk 6]
```diff
@@ -103,6 +103,13 @@ func RunRestore(c context.Context, g glue.Glue, cmdName string, cfg *RestoreConf
 	if err != nil {
 		return err
 	}
+
+	defer func() {
+		if err != nil {
+			log.Error("restore failed", zap.Error(err), zap.Reflect("meta", backupMeta))
```
Comment (kennytm) [inline pkg/task/restore.go]: (ditto)

**Comment by kennytm** [comment]
/run-integration-tests tikv=release-4.0 pd=release-4.0 tidb=release-4.0

**Comment by overvenus** [comment]
/run-integration-tests tikv=release-4.0 pd=release-4.0 tidb=release-4.0

**Comment by kennytm** [comment]
/run-integration-tests tikv=release-4.0 pd=release-4.0 tidb=release-4.0

**Comment by kennytm** [comment]
/run-integration-tests tikv=release-4.0 pd=release-4.0 tidb=release-4.0

**Pull request merged**